deps/build.jl

```diff
@@ -0,0 +1,77 @@
+using Libdl
+
+function find_lib_ker()
+    if haskey(ENV,"JULIA_MATHLINK") && haskey(ENV,"JULIA_MATHKERNEL")
+        return ENV["JULIA_MATHLINK"], ENV["JULIA_MATHKERNEL"]
+    elseif Sys.isapple()
+        # we query OS X metadata for possible non-default installations
+        # TODO: can use `mdls -raw -name kMDItemVersion $path` to get the version
+                
+        # Mathematica
+        for path in readlines(`mdfind "kMDItemCFBundleIdentifier == 'com.wolfram.Mathematica'"`)
+            lib = joinpath(path,"Contents/Frameworks/mathlink.framework/mathlink")
+            ker = joinpath(path,"Contents/MacOS/MathKernel")
+            if isfile(lib) && isfile(ker)
+                return lib, ker
+            end
+        end
+
+        # Wolfram Engine
+        for path in readlines(`mdfind "kMDItemCFBundleIdentifier == 'com.wolfram.engine'"`)
+            # kernels are located in sub-application
+            subpath = joinpath(path, "Contents/Resources/Wolfram Player.app")
+            lib = joinpath(subpath,"Contents/Frameworks/mathlink.framework/mathlink")
+            ker = joinpath(subpath,"Contents/MacOS/MathKernel")
+            if isfile(lib) && isfile(ker)
+                return lib, ker
+            end
+        end
+
+    elseif Sys.isunix()
+        archdir = Sys.ARCH == :arm ?    "Linux-ARM" :
+                  Sys.ARCH == :x86_64 ? "Linux-x86-64" :
+                                        "Linux"
+
+        # alternatively, "math" or "wolfram" is often in PATH, so could use
+        # echo \$InstallationDirectory | math | sed -n -e 's/Out\[1\]= //p'
+
```

Comment:
couldn't we use `mpath =  run(`which mathematica`)` ?

Reply:
That's a good idea.